bindings/wasm/lib/import-model.ts

```diff
@@ -0,0 +1,231 @@
+}
+
+const importers: Array<Importer> = [importGLTF];
+
+const id2node = new Map<number, GLTFTransform.Node>();
+const id2properties = new Map<number, Properties>();
+const id2document = new Map<number, GLTFTransform.Document>();
+
+export const cleanup = () => {
+  id2document.clear();
+  id2node.clear();
+  id2properties.clear();
+};
+
+/**
+ * @internal
+ */
+export const getPropertiesByID = (runID: number) => id2properties.get(runID);
+
+/**
+ * @internal
+ */
+export const getDocumentByID = (runID: number) => id2document.get(runID);
+
+export function getImporterByExtension(extension: string) {
+  const hasExtension =
+      (format: Format) => [format.extension, `.${format.extension}`].includes(
+          extension);
+  const importer = importers.find(im => im.supportedFormats.find(hasExtension));
+
+  if (!importer) {
+    const extensionList =
+        importers
+            .map(importer => importer.supportedFormats.map(f => f.extension))
+            .reduce((acc, cur) => ([...acc, ...cur]))
+            .map(ext => `\`.${ext}\``)
+            .reduceRight(
+                (prev, cur, index) => cur + (index ? ', or ' : ', ') + prev);
+    throw new Error(
+        `Cannot import \`${extension}\`.  ` +
+        `Format must be one of ${extensionList}`);
+  }
+  return importer;
+}
+
+/**
+ *
+ * @group Modelling Functions
+ * @param uri
+ * @returns
+ */
+export async function importModel(uri: string): Promise<NonManifoldGLTFNode> {
+  const importer = importers[0];
+  const sourceDoc = await importer.fetchModel(uri);
+
+  const [sourceNode] = sourceDoc.getRoot().listNodes();
+  if (!sourceNode) {
+    throw new Error(`Model imported from \`${uri}\` contains no nodes.`);
+  }
+  clearNodeTransform(sourceNode);
+
+  const targetNode = new NonManifoldGLTFNode(sourceDoc, sourceNode);
+  targetNode.name = sourceNode.getName();
+  return targetNode;
+}
+
+/**
+ *
+ * @group Modelling Functions
+ * @param uri
+ * @returns
+ */
+export async function importManifold(uri: string): Promise<Manifold> {
+  const sourceNode = await importModel(uri);
+  try {
+    return sourceNode.makeManifold();
+  } catch (e) {
+    const newError = new Error(
+        `Model imported from \`${uri}\` contains no manifold geometry.`);
+    newError.cause = e;
+    throw newError;
+  }
+}
+
+/**
+ * Convert a gltf-transform Node and its children into a Manifold object.
+ *
+ * The original imported model may consist of an entire tree of nodes, each of
+ * which may or may not be manifold.  This method will convert each child node,
+ * and then union the results together.  If a child node has no mesh, the mesh
+ * has no geometry, or the mesh is not manifold, that child node will be
+ * silently excluded.
+ *
+ * Other errors will be re-thrown for the caller to handle.
+ *
+ * @returns A valid Manifold object.
+ */
+export function gltfNodeToManifold(
+    document: GLTFTransform.Document, node: GLTFTransform.Node): Manifold {
```

Comment:
Can we add a time argument? An animated glTF will yield very different manifolds at different times.